api/handlers_images.go

```diff
@@ -144,3 +144,48 @@ func (s *server) ImageUpdateHandler(w http.ResponseWriter, r *http.Request) {
 
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func (s *server) ImageCreateHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := s.mapAccountNumber(vars["account"])
+
+	req := &Ec2ImageCreateRequest{}
+	if err := json.NewDecoder(r.Body).Decode(req); err != nil {
+		msg := fmt.Sprintf("cannot decode body into update image input: %s", err)
+		handleError(w, apierror.New(apierror.ErrBadRequest, msg, err))
+		return
+	}
+
+	if req.InstanceId == nil || req.Name == nil || req.CopyTags == nil || req.ForceReboot == nil {
```

Comment:
Only InstanceId and Name are currently required. We can set defaults for the others if they're not defined.
see https://git.yale.edu/spinup/ec2-api/blob/master/lib/json_schemas/create_ami.json